### PR TITLE
Mo/sdk 756

### PR DESCRIPTION
**Pull request by MarcoOl94**
## Description
Update Bcrypt library.

## Jira Ticket
[756](https://horizenlabs.atlassian.net/jira/software/c/projects/SDK/boards/14?modal=detail&selectedIssue=SDK-756&assignee=615f31c32f6aed0068c67291)

## Changes

- Updated Bcrypt library
- New Unit Test

## Breaking Changes
- No breaking changes, the new library allows to use older version of algorithm

## Checks
- [✔] Project Builds
- [✔] Project passes unit tests
- [✔] Updated documentation accordingly

### CODE REVIEW

File: src/test/scala/sparkz/core/api/http/PeersApiRouteSpec.scala [hunk 1]
```diff
@@ -38,7 +38,7 @@ class PeersApiRouteSpec extends AnyFlatSpec
   private val body = HttpEntity("localhost:8080".asJson.toString).withContentType(ContentTypes.`application/json`)
   private val badBody = HttpEntity("badBodyContent".asJson.toString).withContentType(ContentTypes.`application/json`)
 
-  private val restApiSettingsWithApiKey = RESTApiSettings(addr, Some(BCrypt.hashpw(credentials.password(), BCrypt.gensalt())), None, 10 seconds)
+  private val restApiSettingsWithApiKey = RESTApiSettings(addr, Some(BCrypt.`with`(BCrypt.Version.VERSION_2Y).hashToString(12, credentials.password().toCharArray)), None, 10 seconds)
```
Comment (i-Alex) [inline src/test/scala/sparkz/core/api/http/PeersApiRouteSpec.scala]: what is the meaning of `12` ?

Comment (MarcoOl94) [reply src/test/scala/sparkz/core/api/http/PeersApiRouteSpec.scala]: It is the cost factor of the algorithm which is suggested to be >= 10

Comment (i-Alex) [reply src/test/scala/sparkz/core/api/http/PeersApiRouteSpec.scala]: let's add a `val factor = 12` with some comments what is it.
then use it in this test

File: src/main/scala/sparkz/core/api/http/ApiDirectives.scala [hunk 2]
```diff
@@ -41,7 +41,7 @@ trait ApiDirectives extends CorsHandler with SparkzEncoding {
         if (apiKeyHash.equals(""))
           false
         else
-          BCrypt.checkpw(apiKey, apiKeyHash)
+          BCrypt.verifyer().verify(apiKey.toCharArray, apiKeyHash).verified
```
Comment (i-Alex) [inline src/main/scala/sparkz/core/api/http/ApiDirectives.scala:44]: let's by sure that this method doesn't throw exception.
For example:
1. apiKeyHash is dummy string
2. "apiKey.toCharArray" by sure that it is using UTF8
and so on.

**Review by i-Alex — APPROVED** [review]

**Review by paologalligit — APPROVED** [review]
LGTM

**Pull request merged**